shared/chat/inbox/container/remote.js

```diff
@@ -26,47 +17,57 @@ export type RemoteConvMeta = $Diff<
-      return {
-        backgroundColor: Styles.globalColors.white,
-        channelname: m.channelname,
-        conversationIDKey: m.conversationIDKey,
-        hasBadge: Constants.getHasBadge(state, m.conversationIDKey),
-        hasResetUsers: !!m.resetParticipants && m.resetParticipants.size > 0,
-        hasUnread,
-        iconHoverColor: styles.iconHoverColor,
-        isFinalized: !!m.wasFinalizedBy,
-        isInWidget: true,
-        isMuted: m.isMuted,
-        isSelected: false,
-        // excluding onSelectConversation
-        participantNeedToRekey,
-        participants: m.teamname ? [] : Constants.getRowParticipants(m, _username).toArray(),
-        showBold: styles.showBold,
-        snippet: m.snippet,
-        snippetDecoration: m.snippetDecoration,
-        subColor: styles.subColor,
-        teamname: m.teamname,
-        timestamp: Constants.timestampToString(m),
-        usernameColor: styles.usernameColor,
-        youAreReset: m.membershipType === 'youAreReset',
-        youNeedToRekey,
-      }
-    }).toArray()
+// To cache the list
+const valuesCached = memoize((...vals) => vals.map(v => v))
+
+const metaMapToFirstValues = memoize(metaMap =>
+  metaMap
+    .partialSort(maxShownConversations, (a, b) => b.timestamp - a.timestamp)
+    .filter((_, id) => Constants.isValidConversationIDKey(id))
+    .valueSeq()
+    .toArray()
 )
 
-// Just to cache the sorted values
-const GetNewestConvMetas = createShallowEqualSelector([getSortedConvMetas], map => map)
+// A hack to store the state so we can convert at the last possible minute. This is a lot simpler than plumbing this all the way through
+let _lastState: TypedState
+export const conversationsToSend = (state: TypedState) => {
+  _lastState = state
+  return valuesCached(...metaMapToFirstValues(state.chat2.metaMap))
+}
 
-export default GetNewestConvMetas
+export const serialize = (m: ChatTypes.ConversationMeta): RemoteConvMeta => {
+  const hasUnread = Constants.getHasUnread(_lastState, m.conversationIDKey)
+  const styles = Constants.getRowStyles(m, false, hasUnread)
+  const participantNeedToRekey = m.rekeyers.size > 0
+  const _username = _lastState.config.username || ''
+  const youNeedToRekey = !!participantNeedToRekey && m.rekeyers.has(_username)
+  return {
+    backgroundColor: Styles.globalColors.white,
+    channelname: m.channelname,
+    conversationIDKey: m.conversationIDKey,
+    hasBadge: Constants.getHasBadge(_lastState, m.conversationIDKey),
+    hasResetUsers: !!m.resetParticipants && m.resetParticipants.size > 0,
+    hasUnread,
+    iconHoverColor: styles.iconHoverColor,
+    isFinalized: !!m.wasFinalizedBy,
+    isInWidget: true,
+    isMuted: m.isMuted,
+    isSelected: false,
+    // excluding onSelectConversation
+    participantNeedToRekey,
+    participants: m.teamname
+      ? []
+      : Constants.getRowParticipants(m, _username)
+          .toArray()
+          .slice(0, 2),
```

Comment:
Why the `slice` here?

Reply:
we only need to show 2 people (its for the avatars only)

Reply:
It also truncates the list of participants for me

Reply:
oh i guess its for usernames too, k